frontend/public/components/nav/PinnedResource.tsx

```diff
@@ -0,0 +1,146 @@
+import * as React from 'react';
+import { useDrag, useDrop } from 'react-dnd';
+import { useTranslation } from 'react-i18next';
+import * as classNames from 'classnames';
+import { Button } from '@patternfly/react-core';
+import { MinusCircleIcon, GripVerticalIcon } from '@patternfly/react-icons';
+import { K8sModel, modelFor } from '../../module/k8s';
+import confirmNavUnpinModal from './confirmNavUnpinModal';
+import { NavLinkComponent, ResourceClusterLink, ResourceNSLink, RootNavLink } from './items';
+
+import './PinnedResource.scss';
+
+type PinnedResourceProps = {
+  resourceRef?: string;
+  navResources?: string[];
+  onChange?: (pinnedResources: string[]) => void;
+  idx?: number;
+  draggable?: boolean;
+};
+
+type DraggableButtonProps = {
+  dragRef?: (node) => void | null;
+};
+
+type RemoveButtonProps = {
+  resourceRef?: string;
+  navResources?: string[];
+  onChange?: (pinnedResources: string[]) => void;
+};
+
+export type DragItem = {
+  idx: number;
+  id: string;
+  type: string;
+};
+
+const DraggableButton: React.FC<DraggableButtonProps> = ({ dragRef }) => {
+  const { t } = useTranslation();
+  return (
+    <Button
+      ref={dragRef}
+      className="oc-pinned-resource__drag-button"
+      variant="link"
+      type="button"
+      aria-label={t('public~Drag to reorder')}
+    >
+      <GripVerticalIcon className="oc-pinned-resource__drag-icon" />
+    </Button>
+  );
+};
+
+const RemoveButton: React.FC<RemoveButtonProps> = ({ resourceRef, navResources, onChange }) => {
+  const { t } = useTranslation();
+  const unPin = (e: React.MouseEvent<HTMLButtonElement>, navItem: string) => {
+    e.preventDefault();
+    e.stopPropagation();
+    confirmNavUnpinModal(navItem, navResources, onChange);
+  };
+  return (
+    <Button
+      className="oc-pinned-resource__unpin-button"
+      variant="link"
+      aria-label={t('public~Unpin')}
+      onClick={(e) => unPin(e, resourceRef)}
+    >
+      <MinusCircleIcon className="oc-pinned-resource__delete-icon" />
+    </Button>
+  );
+};
+
+const swapArrayElements = (arr: string[], idx1: number, idx2: number) => {
+  const arrTemp = [...arr];
+  [arrTemp[idx1], arrTemp[idx2]] = [arrTemp[idx2], arrTemp[idx1]];
+  return arrTemp;
+};
```

Comment:
Oh sorry for missing this point the complete time. :man_facepalming: 

As discussed, I'm sure we should not want swap two items, instead of that the user wants to move it up and down between two other resources. Similar to the environment list on a Deployment or Pod; or like trello does this. :smile:

Reply:
updated the GIF